front/src/utils/date.ts

```diff
@@ -229,3 +229,18 @@ export function isArrivalDateInSearchTimeWindow(
   const arrivalDate = new Date(arrivalTime);
   return arrivalDate >= searchDatetimeWindow.begin && arrivalDate <= searchDatetimeWindow.end;
 }
+
+/**
+ * Generates an ISO arrival date string based on the provided date, hours, and minutes.
+ * @param {Date} date - The base date.
+ * @param {{ hours: number; minutes: number }} - An object containing the hours and the minutes of the arrival time.
+ * @returns {string} The ISO formatted arrival date string.
+ */
+export const generateISOArrival = (
```

Comment:
I think the function could be used in contexts other than just arrival dates, so I suggest renaming it to something more generic. What do you think?
```suggestion
export const generateISODateFromDateTime = (
```